lib/src/codepage.dart

```diff
@@ -0,0 +1,486 @@
+    "ΐΑΒΓΔΕΖΗΘΙΚΛΜΝΞΟΠΡ\uFFFDΣΤΥΦΧΨΩΪΫάέήί"
+    "ΰαβγδεζηθικλμνξοπρςστυφχψωϊϋόύώ\uFFFD";
+
+// Characters in ISO-8859-8 above the ASCII and top control characters.
+const _top8859_8 = "\xa0\uFFFD¢£¤¥¦§¨©×«¬\xad®¯°±²³´µ¶·¸¹÷»¼½¾\uFFFD"
+    "\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD"
+    "\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD"
+    "\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD"
+    "\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD‗"
+    "\u05d0\u05d1\u05d2\u05d3\u05d4\u05d5\u05d6\u05d7"
+    "\u05d8\u05d9\u05da\u05db\u05dc\u05dd\u05de\u05df"
+    "\u05e0\u05e1\u05e2\u05e3\u05e4\u05e5\u05e6\u05e7"
+    "\u05e8\u05e9\u05ea\uFFFD\uFFFD\u200e\u200f\uFFFD";
+
+// Characters in ISO-8859-9 above the ASCII and top control characters.
+const _top8859_9 = "\xa0¡¢£¤¥¦§¨©ª«¬\xad®¯°±²³´µ¶·¸¹º»¼½¾¿"
+    "ÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏĞÑÒÓÔÕÖ×ØÙÚÛÜİŞß"
+    "àáâãäåæçèéêëìíîïğñòóôõö÷øùúûüışÿ";
+
+// Characters in ISO-8859-10 above the ASCII and top control characters.
+const _top8859_10 = "\xa0ĄĒĢĪĨĶ§ĻĐŠŦŽ\xadŪŊ°ąēģīĩķ·ļđšŧž―ūŋ"
+    "ĀÁÂÃÄÅÆĮČÉĘËĖÍÎÏÐŅŌÓÔÕÖŨØŲÚÛÜÝÞß"
+    "āáâãäåæįčéęëėíîïðņōóôõöũøųúûüýþĸ";
+
+// Characters in ISO-8859-11 above the ASCII and top control characters.
+const _top8859_11 = "\xa0กขฃคฅฆงจฉชซฌญฎฏฐฑฒณดตถทธนบปผฝพฟ"
+    "ภมยรฤลฦวศษสหฬอฮฯะัาำิีึืฺุู\uFFFD\uFFFD\uFFFD\uFFFD฿"
+    "เแโใไๅๆ็่้๊๋์ํ๎๏๐๑๒๓๔๕๖๗๘๙๚๛\uFFFD\uFFFD\uFFFD\uFFFD";
+
+// Characters in ISO-8859-13 above the ASCII and top control characters.
+const _top8859_13 = "\xa0”¢£¤„¦§Ø©Ŗ«¬\xad®Æ°±²³“µ¶·ø¹ŗ»¼½¾æ"
+    "ĄĮĀĆÄÅĘĒČÉŹĖĢĶĪĻŠŃŅÓŌÕÖ×ŲŁŚŪÜŻŽß"
+    "ąįāćäåęēčéźėģķīļšńņóōõö÷ųłśūüżž’";
+
+// Characters in ISO-8859-14 above the ASCII and top control characters.
+const _top8859_14 = "\xa0Ḃḃ£ĊċḊ§Ẁ©ẂḋỲ\xad®ŸḞḟĠġṀṁ¶ṖẁṗẃṠỳẄẅṡ"
+    "ÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏŴÑÒÓÔÕÖṪØÙÚÛÜÝŶß"
+    "àáâãäåæçèéêëìíîïŵñòóôõöṫøùúûüýŷÿ";
+
+// Characters in ISO-8859-15 above the ASCII and top control characters.
+const _top8859_15 = "\xa0¡¢£€¥Š§š©ª«¬\xad®¯°±²³Žµ¶·ž¹º»ŒœŸ¿"
+    "ÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖ×ØÙÚÛÜÝÞß"
+    "àáâãäåæçèéêëìíîïðñòóôõö÷øùúûüýþÿ";
+
+// Characters in ISO-8859-16 above the ASCII and top control characters.
+const _top8859_16 = "\xa0ĄąŁ€„Š§š©Ș«Ź\xadźŻ°±ČłŽ”¶·žčș»ŒœŸż"
+    "ÀÁÂĂÄĆÆÇÈÉÊËÌÍÎÏĐŃÒÓÔŐÖŚŰÙÚÛÜĘȚß"
+    "àáâăäćæçèéêëìíîïđńòóôőöśűùúûüęțÿ";
+
+const _noControls = "\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD"
+    "\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD"
+    "\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD"
+    "\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD";
+
+// ASCII characters without control characters. Shared by many code pages.
+const _ascii = "$_noControls"
+    r""" !"#$%&'()*+,-./0123456789:;<=>?"""
+    r"@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_"
+    "`abcdefghijklmnopqrstuvwxyz{|}~\uFFFD";
+
+/// A mapping between bytes and characters.
+///
+/// A code page is a way to map bytes to character.
+/// As such, it can only represent 256 different characters.
+abstract class CodePage extends Encoding {
+  /// Decoding converter for code page.
```

Comment:
Not really necessary, the inherited docs from `Encoding` seems sufficient.
I'll remove the comments (and keep the declarations because they return a specialized type).